src/hts/bgzf.nim

```diff
@@ -55,13 +55,19 @@ iterator items*(b: BGZ): string =
   ## iterates over the file line by line
   var 
     kstr: kstring_t
+    r: int
 
   kstr.l = 0
   kstr.m = 0
   kstr.s = nil
   var p = kstr.addr
-  
-  while b.read_line(p) >= 0:
+
+  r = b.read_line(p)
+  while r >= 0:
     yield $kstr.s
+    r = b.read_line(p)
+  
+  if r <= -2:
```

Comment:
this is a nitpick, but could you free before this so that catching the exception won't result in a leak?

Reply:
Done. A leak, is a leak 🤓